astropy/table/column.py

```diff
@@ -301,6 +308,32 @@ def __reduce__(self):
 
         return reconstruct_func, reconstruct_func_args, state
 
+    def get_item(self, item):
+        '''
+        Return self[item] and adjust indices as necessary.
+
+        Parameters
+        ----------
+        item : int, slice, list, or ndarray
+            Element of column to retrieve
+
+        Notes
+        -----
+        This method is an alternative to __getitem__ in the
+        case that index updating is important. The rationale
+        behind creating separate methods is that __getitem__
+        takes a major performance hit if it overrides the
+        (fast) __getitem__ method of ndarray.
+        '''
+        if isinstance(item, INTEGER_TYPES):
+            return self.data[item]  # Return as plain ndarray or ma.MaskedArray
+
+        col_slice = super(BaseColumn, self).__getitem__(item)
+        if isinstance(col_slice, BaseColumn):
```

Comment:
I think in this case we can get away with writing just `if col_slice.__class__ is self.__class__`.  Normally I'd turn my nose up at at that.  But the purpose of this check if I understand correctly is that a slice or some other type of mask was taken of a Numpy array (or subclass thereof).  So I think we can make the assumption that that will always return an instance of the same class that the slice/mask was taken from (unless we also need to support some kind of mixin column where that assumption doesn't hold?  But in that case the mixin could override this code).

In any case, assuming we can get away with it, using `is` will be much faster here.